From 27f50a66124054518d5febe984bccad02ee0846b Mon Sep 17 00:00:00 2001
From: Miguel Arroz <750683+arroz@users.noreply.github.com>
Date: Thu, 2 Sep 2021 18:59:19 -0700
Subject: [PATCH 1/2] #6028: Check if `threadstate->error_t.message` is not
 `git_buf__initbuf` before freeing.

This follows the same principle as `buffer.c` where the same check is done before freeing the buffer. It fixes the crash described in #6028.
---
 src/threadstate.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/threadstate.c b/src/threadstate.c
index 6031e8280..e2c08975f 100644
--- a/src/threadstate.c
+++ b/src/threadstate.c
@@ -36,7 +36,8 @@ static void threadstate_dispose(git_threadstate *threadstate)
 	if (!threadstate)
 		return;
 
-	git__free(threadstate->error_t.message);
+    if (threadstate->error_t.message != git_buf__initbuf)
+        git__free(threadstate->error_t.message);
 	threadstate->error_t.message = NULL;
 }
 
-- 
2.33.0


From 62ee779ea4cdd877419571442860a0d29896a59c Mon Sep 17 00:00:00 2001
From: lhchavez <lhchavez@lhchavez.com>
Date: Sat, 4 Sep 2021 18:01:10 -0700
Subject: [PATCH 2/2] remote: Mark `git_remote_name_is_valid` as `GIT_EXTERN`

This change makes `git_remote_name_is_valid` be part of the public
interface of the library. This is needed for other language bindings to
be able to find this symbol (like in git2go, when linking against
libgit2 dynamically).
---
 include/git2/remote.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/git2/remote.h b/include/git2/remote.h
index 1f52fcd94..51a7d1cdc 100644
--- a/include/git2/remote.h
+++ b/include/git2/remote.h
@@ -971,7 +971,7 @@ GIT_EXTERN(int) git_remote_rename(
  * @param remote_name name to be checked.
  * @return 0 on success or an error code
  */
-int git_remote_name_is_valid(int *valid, const char *remote_name);
+GIT_EXTERN(int) git_remote_name_is_valid(int *valid, const char *remote_name);
 
 /**
 * Delete an existing persisted remote.
-- 
2.33.0

